From 12a9e173624ea368a536517f854519870121076d Mon Sep 17 00:00:00 2001
From: Sachin Prabhu <sprabhu@redhat.com>
Date: Mon, 28 Jul 2014 12:16:45 -0400
Subject: [fs] cifs: Return correct error on query of xattr on file with empty xattrs

Message-id: <1406549822-10376-53-git-send-email-sprabhu@redhat.com>
Patchwork-id: 86734
O-Subject: [RHEL7.1 PATCH 52/69] [bz1082049] Return correct error on query of xattr on file with empty xattrs
Bugzilla: 1082049
RH-Acked-by: J. Bruce Fields <bfields@redhat.com>
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

From: Steve French <smfrench@gmail.com>

xfstest 020 detected a problem with cifs xattr handling.  When a file
had an empty xattr list, we returned success (with an empty xattr value)
on query of particular xattrs rather than returning ENODATA.
This patch fixes it so that query of an xattr returns ENODATA when the
xattr list is empty for the file.

Signed-off-by: Steve French <smfrench@gmail.com>
Reviewed-by: Jeff Layton <jlayton@redhat.com>
(cherry picked from commit 60977fcc808664f82412bb37da7be17640ba99d9)

Signed-off-by: Sachin Prabhu <sprabhu@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/cifs/cifssmb.c b/fs/cifs/cifssmb.c
index f3264bd..6ce4e09 100644
--- a/fs/cifs/cifssmb.c
+++ b/fs/cifs/cifssmb.c
@@ -6197,6 +6197,9 @@ QAllEAsRetry:
  cifs_dbg(FYI, "ea length %d\n", list_len);
  if (list_len <= 8) {
   cifs_dbg(FYI, "empty EA list returned from server\n");
+  /* didn't find the named attribute */
+  if (ea_name)
+   rc = -ENODATA;
   goto QAllEAsOut;
  }
 
-- 
1.7.1